sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/azure/spring/cloud/config/refresh/CalculatedBackoffTime.java

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.config.refresh;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import com.azure.spring.cloud.config.properties.AppConfigurationProviderProperties;
+
+/**
+ * Calculates the amount of time to the next refresh, if a refresh fails.
+ */
+public final class CalculatedBackoffTime {
+
+    static int attempts = 0;
+
+    /**
+     * Sets the number of retry attempts back to 0.
+     */
+    public static void resetAttempts() {
+        attempts = 0;
+    }
+
+    /**
+     * Adds one to the refresh attempt. Should be called every time a refresh attempt fails to increase the backoff
+     * time.
+     */
+    public static void addAttempt() {
+        attempts += 1;
+    }
+
+    /**
+     * Calculates the amount of time to the next refresh, if a refresh fails.
+     * 
+     * @param interval the Refresh Interval
+     * @param properties App Configuration Provider Properties
+     * @return new Refresh Date
+     */
+    public static Instant calculate(Duration interval, AppConfigurationProviderProperties properties) {
+        if (interval == null) {
+            return null;
+        }
+        int durationPeriod = Math.toIntExact(interval.getSeconds());
+
+        Instant now = Instant.now();
+
+        if (durationPeriod <= properties.getDefaultMinBackoff()) {
+            return now.plusSeconds(interval.getSeconds());
+        }
+
+        int defaultMinBackoff = properties.getDefaultMinBackoff();
+        double min = Math.min(Math.pow(2, attempts - 1), durationPeriod);
+
+        long maxBackoff = Math.min(interval.getSeconds(), properties.getDefaultMaxBackoff());
```

Comment:
Change usage of seconds to nanoseconds. For both interval and default max Backoff.